text/0000-buildpack-registry-search-api.md

```diff
@@ -0,0 +1,242 @@
+# Meta
+[meta]: #meta
+- Name: Buildpack Registry Search API
+- Start Date: 2020-11-11
+- Author(s): @elbandito
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+This is a proposal for a service API that enables clients to search for buildpacks.
+
+# Definitions
+[definitions]: #definitions
+
+**API** - Application Programmer Interface
+
+**Buildpack Registry** - the CNB centeralized [registry](https://github.com/buildpacks/registry-index) that contains metadata for all the community supported buildpacks 
+
+**Service** - an independent web process running in the cloud that provides RESTful endpoints
+
+**Client** - represents a physical end-user or software interface e.g. CLI, web UI, etc.
+ 
+# Motivation
+[motivation]: #motivation
+
+As the CNB project makes progress towards supporting a centralized Buildpack Registry, I anticipate the need to quickly search for and identify specific buildpacks.  To support this, I propose the creation of an API search service that customers could use directly and/or indirectly via a web dashboard or CLI.
+
+- What use cases does it support?  
+  - Customers looking to quickly indentify usefulful buildpacks to build their applications.
+- What is the expected outcome?  
+  - Customers will be able to quickly find relevant buildpacks for their use cases.
+
+# What it is
+[what-it-is]: #what-it-is
+
+An external API service that exposes endpoints for retrieving buildpack metadata against the official CNB buildpack registry.  It's NOT the goal of this API to support private buildpack registries.  I envision this service providing a few GET endpoints to support plain text query searches, and individual buildpack data retrieval.  The API endpoints will be versioned, and follow OpenAPI and/or Json Schema/API standards.
+
+### Versioned Endpoints
+
+All the endpoints defined below will be version via a `vN` prefix to the base URL. e.g. `https://registry.buildpacks.io/api/v1/<endpoint>`.
+
+### Supported MIME types
+
+`Accept: application/vnd.buildpacks+json`
```

Comment:
awesome to see this. I imagine in the future we'll probably do something similar to what [GitHub does](https://docs.github.com/en/free-pro-team@latest/rest/overview/media-types).